### ISSUE TITLE
supporting rancher 

### ISSUE THREAD

**Issue by lavvy** [hypriot/cluster-lab#9]
Is there any plans to support the rancher project

**Comment by luxas** [comment]
No, not in the near future. Because RancherOS won't run on ARM.
I've worked on this earlier getting RancherOS to ARM earlier this autumn, see https://github.com/rancher/os/issues/417
Now I'm working on [kubernetes-on-arm](https://github.com/luxas/kubernetes-on-arm) and core Kubernetes instead.

However, it is needed more work before it may run on ARM, feel free to fork my RancherOS forks, if you want to hack on it.
So I don't think the Hypriot guys will support something that isn't supported by Rancher or even present.

But if you want to make it happen, read the thread and start to hack! :smile:

**Comment by lavvy** [comment]
I had been thinking, but rancher agent do run on other OSes,  cant we have
hypriotos as another os to rancher agent,  in that case hypriotos too could
be used as another host like ubuntu and coreos.

**Comment by luxas** [comment]
Seems like `rancher` uses java, python and some stuff.
Test to cross-compile if you want, it's not impossible at all
It was unfortunate that it wasn't built with `go`, then it would have been easier.

You could create an issue in `rancher/rancher` and see what they say.

**Comment by DieterReuter** [comment]
@lavvy What are your use cases for running Ranger on HypriotOS? I just like to figure out what parts do you want to run on HypriotOS. If it's just the agent, I think it could be done easier as a first step. And it would make sense too, to control a fleet of HypriotOS machines from a central Ranger system.
On the other hand, the Ranger guys are already working on it to port all their stack on ARM - so I can clearly imagine it would be a good thing to have at least a Ranger Agent first running on ARM and on HypriotOS as well.

And as long as the Ranger Agent is just a Docker Image, one has to do the work in porting it to ARMv6 or ARMv7. Then it should be run on all Docker hosts with ARM arch, not only on HypriotOS. I would prefer this too, supporting as flavours of ARM OSes like Ubuntu/Debian/Raspbian/Arch/HypriotOS makes even more sense. 

This work should be merged into upstream at `ranger/ranger`, so I totally agree with @luxas. Just file an issue at Ranger and help them to get it done. /cc @ibuildthecloud, imikushin

**Comment by lavvy** [comment]
Rancher is splendid with an app store and all the pipes are exposed in
rancher.  It has all major apps like hadoop,  kubernates etc in its app
store or catalog.  So if you have rancher,  you have it all and with smooth
upgradeability.  If we have rancher agent and rancher server ported to
arm,  we can create an arm appstore for arm compatible apps like kubernates
etc and can be upgraded with ease too

**Comment by dbones** [comment]
hey, here is the related issue in the rancher project
https://github.com/rancher/rancher/issues/2101
